components/treelist/selection/cells.md

```diff
@@ -0,0 +1,253 @@
+    {
+        for (int i = 1; i <= 15; i++)
+        {
+            TreeListData.Add(new Employee()
+                {
+                    Id = i,
+                    ParentId = i <= 3 ? null : i % 3 + 1,
+                    FirstName = "First " + i,
+                    LastName = "Last " + i,
+                    Position = i <= 3 ? "Team Lead" : "Software Engineer"
+                });
+        }
+
+        SelectedCells = new List<TreeListSelectedCellDescriptor>() {
+            new TreeListSelectedCellDescriptor()
+            {
+                DataItem = TreeListData.ElementAt(2),
+                ColumnField = nameof(Employee.FirstName)
+            }
+        };
+    }
+
+    public class Employee
+    {
+        public int Id { get; set; }
+        public int? ParentId { get; set; }
+        public string FirstName { get; set; } = string.Empty;
+        public string LastName { get; set; } = string.Empty;
+        public string Position { get; set; } = string.Empty;
+    }
+}
+````
+
+### SelectedCellsChanged and Asynchronous Operations
+
+The `SelectedCellsChanged` event handler cannot be awaited. To execute asynchronous operations when the user selects rows, use the [`OnRowClick`]({%slug treelist-events%}#onrowclick) or [`OnRowDoubleClick`]({%slug treelist-events%}#onrowdoubleclick) event instead.
+
+## TreeListSelectedCellDescriptor
+
+The `TreeListSelectedCellDescriptor` type exposes the following properties:
+
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Property Name | Type | Description |
+| --- | --- | --- |
+| `ColumnField` | `string` | The value of the TreeList column `Field` parameter, if set. |
+| `ColumnId` | `string` | The value of the TreeList column `Id` parameter, if set. |
+| `DataItem` | `object` | The TreeList data item instance. Cast it to the actual TreeList model type before use. |
+
+## Selection When Data Changes
+
+When the TreeList `Data` collection changes, the `SelectedCells` collection has the following behavior:
+
+* When the user updates a selected cell and the item instance is replaced, you have to also replace the `DataItem` object in the `SelectedCells` collection. Do that in the [TreeList `OnUpdate` event]({%slug treelist-editing-overview%}#events).
+* When the user deletes a row with selected cells, update the `SelectedCells` collection in the the TreeList `OnDelete` event handler.
+* To select cells from a new item in the TreeList you can use the [`OnCreate` event]({%slug treelist-editing-overview%}#events) to update the `SelectedCells` collection.
+
+### Equals Comparison
+
+The items in `SelectedCells` are compared against the items in the TreeList data in order to determine which cells will be highlighted. The default framework behavior is to compare objects by reference. The data item references may not match when:
+
+* The `SelectedCells` are obtained from a different data source than the all TreeList items, for example, from a separate service.
```

Comment:
Generally speaking, single item lists are not really lists and should be avoided. What's the case for a single item list here? Is it better visibility and rendering on the page?

```suggestion
* The `SelectedCells` are obtained from a different data source than all the TreeList items, for example, from a separate service.
```